app/Models/Quiz.php

```diff
@@ -42,7 +46,12 @@ public function answers(): HasManyThrough
 
     public function isLocked(): Attribute
     {
-        return Attribute::get(fn(): bool => $this->locked_at !== null);
+        return Attribute::get(fn(): bool => $this->isReadyToBePublished() && $this->scheduled_at <= Carbon::now());
+    }
+
+    public function closeAt(): Attribute
+    {
+        return Attribute::get(fn(): ?Carbon => $this->isReadyToBePublished() ? $this->scheduled_at->copy()->addSeconds($this->duration) : null);
```

Comment:
```suggestion
        return Attribute::get(fn(): ?Carbon => $this->isReadyToBePublished() ? $this->scheduled_at->copy()->addMinutes($this->duration) : null);
```

Minutes should be more manageable for user - it's about calculating minutes into seconds.